### ISSUE TITLE
fs.appendFileSync & fs.appendFile are not working

### ISSUE THREAD

**Issue by ssxpatrick** [oven-sh/bun#564]
Hello,
I'm experiencing the following error on MacOS 12.4 M1:

`fs.appendFileSync` is throwing this error:

    fs.appendFileSync("greeting.txt", "Hello")
    ^
    EBADF: Bad file descriptor
    syscall: "write"
    errno: -9`

`fs.appendFile` is not throwing an error, but no data is being appended to the file. 

Thanks for all the hard work!

**Comment by vitalspace** [comment]
I don't know what the solution to this error is. but you can use this alternative.

```
await Bun.write("output.txt", "Hello World");
```

**Comment by Electroid** [comment]
It appears this has been fixed on MacOS M1, as of Bun [v0.5.0](https://github.com/oven-sh/bun/releases/tag/bun-v0.5.0). If you still encounter this bug, let us know and we'll re-open the issue.

**Comment by huedaya** [comment]
Still experiencing same error while calling `Bun.file()` on `Ubuntu 20.04.3 LTS`. But working while running on MacOS M1.

```
app-bun-test | Listening on http://localhost:3000 ...
app-bun-test | EBADF: Bad file descriptor
app-bun-test |    path: "data.json"
app-bun-test |  syscall: "open"
app-bun-test |    errno: -9
```

Trying to run inside Docker 
```
FROM oven/bun:latest
WORKDIR /app
COPY package.json package.json
# COPY bun.lockb bun.lockb
RUN bun install
COPY . .
EXPOSE 3000
ENTRYPOINT ["bun","--watch", "index.js"]
```
